framework/include/constraints/ADMortarConstraint.h

```diff
@@ -36,6 +36,24 @@ class ADMortarConstraint : public MortarConstraintBase
    */
   void computeJacobian(Moose::MortarType mortar_type) override;
 
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+
+  /**
+   * Get rid of AD derivative entries by dof index
+   */
+  void trimDerivative(dof_id_type remove_derivative_index, ADReal & dual_number);
```

Comment:
since it's pass by value, the caller of this method knows that whatever they pass from the callee will not be modified, so from an API standpoint that is enough. You'll note that in the *implementation* I mark the argument as `const`